### PR TITLE
Adding doc on how to setup Fail2ban

### PR DESCRIPTION
**Pull request by GuillaumeHullin**
## Proposed change

This is a short documentation on how to setup Fail2ban for Paperless. It also include documentation for creating a jail when Paperless is running behind a reverse proxy.

With #3382 fix, detection of IP address with Fail2ban is working. 

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (please explain)

This is only a documentation addition.

## Checklist:

- [ ] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [x] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: docs/administration.md [hunk 1]
```diff
@@ -196,6 +196,75 @@ $ python3 manage.py migrate documents <migration number>
...
+The first step is to create a fail2ban jail configuration file for Paperless. Open your preferred text editor and create a new file named `paperless.conf` in the `/etc/fail2ban/jail.d/` directory.
+
+Add the following contents to the file:
+
+```
+[paperless]
+enabled  = true
+maxretry = 5
+filter   = paperless
+logpath  = /var/lib/docker/volumes/paperless_data/_data/log/paperless.log
```
Comment (stumpylog) [inline docs/administration.md:216]: This seems like a good line to include something (like a note or [annotation](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-annotations)) to adjust the path as necessary, if not using the default volumes or paths.

**Review by stumpylog — CHANGES_REQUESTED** [review]

File: docs/administration.md (hunk 1 above)
Comment (GuillaumeHullin) [reply docs/administration.md]: Done. I have added a warning.

**Comment by shamoon** [comment]
Perhaps this should be in the wiki with a link to it from the docs? Otherwise we're maintaining docs for something a bit outside the scope of -ngx. Not sure what others think

**Comment by GuillaumeHullin** [comment]
@shamoon no worries. I understand why it shouldn't be in the docs. I did created the Wiki page https://github.com/paperless-ngx/paperless-ngx/wiki/Fail2Ban 
Do you want to have a link in the docs? Or should I just remove what I did?

**Comment by shamoon** [comment]
I'd say it's up to you if you'd like to do that, but yes we can include a link in the docs to the wiki so people know it's there. 

Perhaps it could be a little more general though in case others add things like crowdsec etc in the future. I'm not sure what exactly these tools are called as a group, maybe "security tools"? As an example here's the docs now for nginx reverse proxy: https://docs.paperless-ngx.com/setup/#nginx

**Comment by shamoon** [comment]
And how about this: https://github.com/paperless-ngx/paperless-ngx/wiki/Using-Security-Tools-with-Paperless-ngx , so for fail2ban would be https://github.com/paperless-ngx/paperless-ngx/wiki/Using-Security-Tools-with-Paperless-ngx#fail2ban

**Comment by GuillaumeHullin** [comment]
I added a section under the setup of NGINX. I did kept the reference to Fail2ban in order to be searchable (that's what I did originally).

File: docs/setup.md [hunk 2]
```diff
@@ -863,3 +863,6 @@ For details, refer to [configuration](/configuration).
 # Using nginx as a reverse proxy {#nginx}
 
 Please see [the wiki](https://github.com/paperless-ngx/paperless-ngx/wiki/Using-a-Reverse-Proxy-with-Paperless-ngx#nginx) for user-maintained documetation of using nginx with Paperless-ngx.
+
+# Enhancing security {#security}
+Please see [the wiki](https://github.com/paperless-ngx/paperless-ngx/wiki/Using-Security-Tools-with-Paperless-ngx) for user-maintained documetation of how to configure your security tools (like Fail2ban) with Paperless-ngx.
```
Comment (shamoon) [inline docs/setup.md:868]:
```suggestion
Please see [the wiki](https://github.com/paperless-ngx/paperless-ngx/wiki/Using-Security-Tools-with-Paperless-ngx) for user-maintained documentation of how to configure security tools like Fail2ban with Paperless-ngx.
```

**Review by shamoon — CHANGES_REQUESTED** [review]
This is great, thanks! Theres a typo and minor changes to wording

**Comment by shamoon** [comment]
Ha great catch on the other one =)

**Review by shamoon — DISMISSED** [review]

**Review by shamoon — APPROVED** [review]

**Review by stumpylog — APPROVED** [review]

**Pull request merged**